### ISSUE TITLE
Authentications via ldaps

### ISSUE THREAD

**Issue by dermig** [gitbucket/gitbucket#1782]
## Issue
**Impacted version**: 4.18.0

**Deployment mode**: via nginx

**Problem description**:
Hi,

I try to get the login credentials from a Samba Server (running in AD mode). Using "normal" ldap, everything works fine. I followed https://github.com/gitbucket/gitbucket/wiki/LDAP-Authentication-Settings to get it to work with ssl encryption. I imported the certificate from the ldap server via:
echo | openssl s_client -connect host.bla.blubb:636 2>&1 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > test.pem
keytool -import -trustcacerts -file test.pem -alias host.bla.blubb -keystore /path/to/my/keystore
I used the standard java keystore password "changeit"

I configured gitbucket via system settings:
- LDAP host: host.bla.blubb
- LDAP port: 636
- Bind DN: cn=....
- Bind password: pw
- Base DN: dc=...
- User name attribute: uid
- Additional filter condition:
- Full name attribute: cn
- Mail address attribute: mail
- Enable TLS: unchecked
- Enable SSL: checked
- Keystore: /path/to/my/keystore

 Unfortunately, I see the following error in syslog:
java[1259]: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

I would appreciate it if anyone could give me a hint on it.

EDIT:
I successfully tested the connection using https://gist.github.com/4ndrej/4547029 

Regards
Michael

**Comment by uli-heller** [comment]
https://gist.github.com/4ndrej/4547029 contains the same error message as above. Did you take the actions mentioned there?

**Comment by dermig** [comment]
Hi,

please excuse my late reply. It has been working for a while now, I just forgot to mention it here.

Unfortunately, it's difficult for me to say how the problem has been fixed. I use the settings as given in my first post, except that I imported the server certificate via "keytool -importcert -destkeystore ldap.jks -file test.pem".

**Comment by uli-heller** [comment]
OK, thanks. So maybe we should close the issue?

**Comment by dermig** [comment]
> So maybe we should close the issue?
Yes.

**Comment by uli-heller** [comment]
@dermig Would you mind to close it? I cannot do it since I didn't create the issue... Thx, Uli

**Comment by dermig** [comment]
Hi,

I reopened this issue because I realized that apparently, the field "Keystore" in "System Settings/Authentication" is useless: after updating java / installing a new version of java, secure connections to our ldap server no longer worked due to SSL errors. I got rid of the problem after importing the self-signed certificate into the standard keystore of java (normally located in $JAVA_HOME/lib/security/cacerts). Thus, I guess that GitBucket currently only is checking the standard java keystore and ignoring the path which is set by the "Keystore" field in settings.

Regards,
Michael
